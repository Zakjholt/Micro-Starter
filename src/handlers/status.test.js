const test = require('ava')
const micro = require('micro')
const listen = require('test-listen')
const axios = require('axios')
const app = require('../../src/app')

test('status endpoint', async t => {
  const microInstance = micro(app) // Runs an instance of the microservice
  const url = await listen(microInstance) // Finds the url that the service is running on
  const { data } = await axios.get(`${url}/status`) // Sends a request to the running microservice

  t.snapshot(data) // Snapshots the response
})

test('echo endpoint success', async t => {
  const microInstance = micro(app)
  const url = await listen(microInstance)
  const { data: { message } } = await axios.post(`${url}/echo`, {
    message: 'Test'
  })

  t.snapshot(message)
})

test('echo endpoint 400 on validator schema error', async t => {
  const microInstance = micro(app)
  const url = await listen(microInstance)
  const { response: { status } } = await t.throws(
    // Use t.throws to let ava know you're expecting an error to be thrown. Otherwise the test will fail
    axios.post(`${url}/echo`, {
      property: 'Missing the correct property type'
    })
  )

  t.is(status, 400)
})
