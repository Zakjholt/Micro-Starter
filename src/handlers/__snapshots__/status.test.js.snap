// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`echo endpoint success 1`] = `"Test"`;

exports[`status endpoint 1`] = `
Object {
  "status": "healthy",
}
`;
